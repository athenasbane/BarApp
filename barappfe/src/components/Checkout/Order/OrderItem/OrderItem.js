import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Typography, makeStyles, Button } from '@material-ui/core';
import DeleteForeverIcon from '@material-ui/icons/DeleteForever';

const useStyles = makeStyles({
  root: {
    border: '1px solid black',
    textAlign: 'center',
  },
});

const OrderItem = ({ item, removeItem }) => {
  const classes = useStyles();
  return (
    <Grid className={classes.root} container item direction="row" alignItems="center">
      <Grid item xs={4}>
        <Typography>
          {item.title}
          &nbsp;-&nbsp;
          {item.subOption}
        </Typography>
      </Grid>
      <Grid item xs={3}>
        <Typography>{item.volume}</Typography>
      </Grid>
      <Grid item xs={3}>
        <Typography>
          Â£&nbsp;
          {(item.volume * item.price).toFixed(2)}
        </Typography>
      </Grid>
      <Grid item xs={2}>
        <Button onClick={() => removeItem(item)}>
          <DeleteForeverIcon />
        </Button>
      </Grid>
    </Grid>
  );
};

OrderItem.propTypes = {
  item: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.string])).isRequired,
  removeItem: PropTypes.func.isRequired,
};

export default OrderItem;
