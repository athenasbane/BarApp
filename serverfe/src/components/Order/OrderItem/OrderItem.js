import React from 'react';
import {
    Grid, 
    Typography,
    makeStyles
} from '@material-ui/core';

const useStyles = makeStyles({
    root: {
        border: '2px solid black',
        width: '80vw'
    }
})

const OrderItem = props => {
    const classes = useStyles();
    return (
        <Grid className={classes.root} justify="center" container item direction="row" spacing={2}>
            <Grid item>
                <Typography><strong>Title:</strong>&nbsp;{props.item.title}</Typography>
            </Grid>
            <Grid item>
                <Typography><strong>Option:</strong>&nbsp;{props.item.subOption}</Typography>
            </Grid>
            <Grid item>
                <Typography><strong>Volume:</strong>&nbsp;{props.item.volume}</Typography>
            </Grid>
        </Grid>
    );
}

export default OrderItem;