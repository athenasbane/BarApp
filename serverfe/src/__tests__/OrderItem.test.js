import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import OrderItem from '../components/Order/OrderItem/OrderItem';

const item = {
    optionId: "5f6a3d7bf2723782c641bb35",
    subOption: "Half-Pint",
    title: "Strongbow",
    volume: 3,
    _id: "5f7f2a2112516b19cd01ef61",
};

describe('OrderItem Tests', () => {
    test('renders correctly', () => {
        const tree = renderer
            .create(<OrderItem item={item} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});