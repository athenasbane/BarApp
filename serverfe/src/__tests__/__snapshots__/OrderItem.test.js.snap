// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem Tests renders correctly 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <strong>
        Title:
      </strong>
       
      Strongbow
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <strong>
        Option:
      </strong>
       
      Half-Pint
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <strong>
        Volume:
      </strong>
       
      3
    </p>
  </div>
</div>
`;
